src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -2382,16 +2382,38 @@ private bool HasGetChildItemDynamicParameters(ProviderInfo providerInfo)
                         // Since the path contained glob characters or we are recursing and the
```

Comment:
@iSazonov, your last commit had 2 failures in `PowerShell-CI-macos`
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Get-ChildItem -Path -Depth 2 -Name -Force !!!!!!!!
```powershell
Expected 2, but got 1.
at <ScriptBlock>, /Users/runner/runners/2.169.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 369
369:         1 | Should -Be 2
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Depth -Name.Get-ChildItem -Path -Depth 2 -Name -Force
```powershell
Expected 13, but got 12.
at <ScriptBlock>, /Users/runner/runners/2.169.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 359
359:             $result.Count | Should -Be 13
```